README.md

```diff
@@ -257,6 +257,105 @@ replacing the type converters with something else. These elements, in possible
 conjunction with elements from the Hierarchy API, are flexible enough to build
 anything you need.
 
+## Special Type Parsing and Casting
+
+We use the `cast` library for casting values read in from configurations into their go types. The `cast` library 
+falls back to JSON for complex types expressed as string values. Here are some examples of how we parse different types:
+
+**[]string:**
+For a given configuration
+```go
+type Config struct {
+    TheSlice *[]string
+}
+```
+The values in the following examples will all be parsed as a string slice.
+```yaml
+theslice:
```

Comment:
suggestion: This illustrates how the slice part of the config would look but it doesn't represent what the full configuration file would look like if someone used the above `Config` struct. Each setting is always rooted under a key that is named after the `Config` struct. For example, `type Config struct { TheSlice []string }` would end up needing this YAML tree:
```yaml
config:
  theslice:
    - "a"
    - "b"
    - "c"
```

The `Config` can be given a `Name() string` method that changes the name of the root. The default name is the name of the struct otherwise.